app/Models/Team.php

```diff
@@ -281,30 +281,30 @@ public function save(array $options = [])
         }
 
         if (!$this->exists) {
-            $this->channel_id ??= 0;
-            $saved = $this->getConnection()->transaction(function () use ($options) {
+            return $this->getConnection()->transaction(function () use ($options) {
                 return (new Chat\Channel())->getConnection()->transaction(function () use ($options) {
-                    $channel = $this->createChannel();
+                    $this->channel_id ??= 0;
                     $saved = parent::save($options);
 
                     if ($saved) {
                         $this->members()->create(['user_id' => $this->leader_id]);
+
+                        $channel = $this->createChannel();
                         $channel->addUser($this->leader);
+
+                        $this->flag()->updateFile();
+                        $this->header()->updateFile();
                     }
 
-                    return $saved;
+                    return parent::save($options);
```

Comment:
Should probably only need to call `parent::save` an extra time only if some the files are updated; in the case of it returning `false` it's just calling twice in a row to return `false` but that can be fixed later

Reply:
it's also to save the real channel id